stubs/bleach/bleach/sanitizer.pyi

```diff
@@ -1,65 +1,77 @@
-ALLOWED_PROTOCOLS: list[str]
+ALLOWED_PROTOCOLS: frozenset[str]
 
 INVISIBLE_CHARACTERS: str
 INVISIBLE_CHARACTERS_RE: Pattern[str]
 INVISIBLE_REPLACEMENT_CHAR: str
 
 # A html5lib Filter class
-_Filter: TypeAlias = Any
+class _Filter(Protocol):
+    def __call__(self, *, source: BleachSanitizerFilter) -> Incomplete: ...
+
+_AttributeFilter: TypeAlias = Callable[[str, str, str], bool]
+_AttributeDict: TypeAlias = dict[str, list[str] | _AttributeFilter] | dict[str, list[str]] | dict[str, _AttributeFilter]
+_Attributes: TypeAlias = _AttributeFilter | _AttributeDict | list[str]
+
+_TreeWalker: TypeAlias = Callable[[Incomplete], Incomplete]
 
 class Cleaner:
-    tags: Container[str]
+    tags: Iterable[str]
     attributes: _Attributes
-    protocols: Container[str]
+    protocols: Iterable[str]
     strip: bool
     strip_comments: bool
     filters: Iterable[_Filter]
     css_sanitizer: CSSSanitizer | None
     parser: BleachHTMLParser
-    walker: Any
+    walker: _TreeWalker
     serializer: BleachHTMLSerializer
     def __init__(
         self,
-        tags: Container[str] = ...,
+        tags: Iterable[str] = ...,
         attributes: _Attributes = ...,
-        protocols: Container[str] = ...,
+        protocols: Iterable[str] = ...,
         strip: bool = ...,
         strip_comments: bool = ...,
         filters: Iterable[_Filter] | None = ...,
         css_sanitizer: CSSSanitizer | None = ...,
     ) -> None: ...
     def clean(self, text: str) -> str: ...
 
-_AttributeFilter: TypeAlias = Callable[[str, str, str], bool]
-_AttributeDict: TypeAlias = dict[str, list[str] | _AttributeFilter] | dict[str, list[str]] | dict[str, _AttributeFilter]
-_Attributes: TypeAlias = _AttributeFilter | _AttributeDict | list[str]
-
 def attribute_filter_factory(attributes: _Attributes) -> _AttributeFilter: ...
 
 class BleachSanitizerFilter(SanitizerFilter):
+    allowed_tags: frozenset[str]
+    allowed_protocols: frozenset[str]
     attr_filter: _AttributeFilter
-    strip_disallowed_elements: bool
+    strip_disallowed_tags: bool
     strip_html_comments: bool
+    attr_val_is_uri: frozenset[tuple[str | None, str]]
+    svg_attr_val_allows_ref: frozenset[tuple[str | None, str]]
+    svg_allow_local_href: frozenset[tuple[str | None, str]]
+    css_sanitizer: CSSSanitizer | None
     def __init__(
         self,
         source,
-        allowed_elements: Container[str] = ...,
+        allowed_tags: Iterable[str] = ...,
         attributes: _Attributes = ...,
-        allowed_protocols: Container[str] = ...,
-        strip_disallowed_elements: bool = ...,
-        strip_html_comments: bool = ...,
-        css_sanitizer: CSSSanitizer | None = ...,
-        **kwargs,
+        allowed_protocols: Iterable[str] = ...,
+        attr_val_is_uri: frozenset[tuple[str | None, str]] = ...,
+        svg_attr_val_allows_ref: frozenset[tuple[str | None, str]] = ...,
+        svg_allow_local_href: frozenset[tuple[str | None, str]] = ...,
```

Comment:
These are the same type as the `_Attr` alias that you added in `bleach/callbacks.pyi`, and I don't think that's a coincidence. Maybe the `_Attr` alias you added there should be moved to `bleach/__init__.pyi`, renamed as `_HTMLAttr`, and imported here as well?